### PR TITLE
RTCRtpTransceiver: setCodecPreferences - improve docs

### PR DESCRIPTION
**Pull request by hamishwillee**
This is an update to [`RTCRtpTransceiver.setCodecPreferences()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpTransceiver/setCodecPreferences) to capture some of the useful information from the Moz blog https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/

Specifically this makes it clear why you want to fetch the supported codecs first and then reorder them and steals an example from the blog showing how. It also makes it more clear when you call this, and that you can do so after the connection has already been started.

It doesn't cover the things in the blog that are still in discussion/not in spec.

Related to #33981

### CODE REVIEW

**Comment by hamishwillee** [comment]
@bsmth FYI This fell out of you pointing out https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/ to me.

That has more code, and is pretty cool. But the examples don't work for me locally, so I have cut out the relevant information. I might at some point try and do these as live examples - can be a bit tricky but is doable for some cases.

File: files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md [hunk 1]
```diff
@@ -65,31 +76,54 @@ None ({{jsxref("undefined")}}).
...
+```
+
+The method takes the list of supported codecs, and an array containing the preferred MIME types, in decreasing order, and returns the array sorted in place.
+The code below shows how this is used, assuming that you have already set up a peer connection (`peerConnection`):
+
+```js
+// Get supported codecs the sort using preferred codecs
+const supportedCodecs = RTCRtpReceiver.getCapabilities("video").codecs;
+const preferredCodecs = ["video/H264", "video/VP8", "video/VP9"];
+const sortedCodecs = sortByMimeTypes(supportedCodecs, preferredCodecs);
```
Comment (bsmth) [inline files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md:120]: Yeah this is nice, very good addition! 👏🏻

File: files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md [hunk 2]
```diff
@@ -100,6 +134,7 @@ const availReceiveCodecs = transceiver.receiver.getCapabilities("video").codecs;
 
 ## See also
 
+- [setCodecPreferences is now in all browsers!](https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/) in the Mozilla Blog.
```
Comment (bsmth) [inline files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md:137]: I believe this is the recommended format 🤔 

```suggestion
- [setCodecPreferences is now in all browsers!](https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/) on blog.mozilla.org (2024)
```

**Review by bsmth — APPROVED** [review]
Really nice, thank you Hamish! One comment left for See Also format, but ready to go!

File: files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md (hunk 2 above)
Comment (hamishwillee) [reply files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md:137]: Thanks. I had a hand in that guideline, but it seems so long ago. In any case, it's a better format

File: files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md [hunk 3]
```diff
@@ -65,31 +76,54 @@ None ({{jsxref("undefined")}}).
...
-
-The following code snippet demonstrates how to get both the list of codecs supported by the transceiver's {{domxref("RTCRtpSender")}} and {{domxref("RTCRtpReceiver")}}.
+You can get the codecs supported for decoding data using the {{domxref("RTCRtpReceiver.getCapabilities_static", "RTCRtpReceiver.getCapabilities()")}} static method as shown:
 
 ```js
-const availSendCodecs = transceiver.sender.getCapabilities("video").codecs;
 const availReceiveCodecs = transceiver.receiver.getCapabilities("video").codecs;
 ```
 
+To reorder the codecs array to our preferred order, we can use the sorting fuction below to sort on MIME type (this comes from [setCodecPreferences is now in all browsers!](https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/) in the Mozilla Blog).
```
Comment (hamishwillee) [inline files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md:99]:
```suggestion
To reorder the codecs array to our preferred order, we can use the sorting fuction below to sort on MIME type (this comes from [setCodecPreferences is now in all browsers!](https://blog.mozilla.org/webrtc/cross-browser-support-for-choosing-webrtc-codecs/) on blog.mozilla.org (2024)).
```

**Comment by hamishwillee** [comment]
Thank you for the review, and even more for pointing me to the blog!

**Pull request merged**